src/System.Net.Security/tests/FunctionalTests/SslStreamAlpnTests.cs

```diff
@@ -116,34 +118,62 @@ public class SslStreamAlpnTests
-                {
-                    ApplicationProtocols = new List<SslApplicationProtocol> { SslApplicationProtocol.Http2 },
-                    ServerCertificate = certificate,
-                };
-
-                Task t1 = Assert.ThrowsAsync<AuthenticationException>(() => client.AuthenticateAsClientAsync(clientOptions, CancellationToken.None));
-                Task t2 = Assert.ThrowsAsync<AuthenticationException>(() => server.AuthenticateAsServerAsync(serverOptions, CancellationToken.None));
-
-                await TestConfiguration.WhenAllOrAnyFailedWithTimeout(t1, t2);
+                    Task<TcpClient> serverTask = listener.AcceptTcpClientAsync();
+                    await client.ConnectAsync(IPAddress.Loopback, ((IPEndPoint)listener.LocalEndpoint).Port);
+
+                    using (TcpClient server = await serverTask)
+                    using (SslStream serverStream = new SslStream(server.GetStream(), leaveInnerStreamOpen: false))
+                    using (SslStream clientStream = new SslStream(client.GetStream(), leaveInnerStreamOpen: false))
+                    using (X509Certificate2 certificate = Configuration.Certificates.GetServerCertificate())
+                    {
+                        SslServerAuthenticationOptions serverOptions = new SslServerAuthenticationOptions
+                        {
+                            ApplicationProtocols = new List<SslApplicationProtocol> { SslApplicationProtocol.Http2 },
+                            ServerCertificate = certificate,
+                        };
+                        SslClientAuthenticationOptions clientOptions = new SslClientAuthenticationOptions
+                        {
+                            ApplicationProtocols = new List<SslApplicationProtocol> { SslApplicationProtocol.Http11 },
+                            RemoteCertificateValidationCallback = AllowAnyServerCertificate,
+                            TargetHost = certificate.GetNameInfo(X509NameType.SimpleName, false),
+                        };
+
+                        // Test alpn failure only on platforms that supports ALPN.
+                        if ((RuntimeInformation.IsOSPlatform(OSPlatform.Linux) && !(PlatformDetection.IsUbuntu1404 || PlatformDetection.IsDebian8 || PlatformDetection.IsCentos6)) ||
+                            (PlatformDetection.IsWindows && !PlatformDetection.IsWindows7))
+                        {
+                            Task t1 = Assert.ThrowsAsync<IOException>(() => clientStream.AuthenticateAsClientAsync(clientOptions, CancellationToken.None));
+                            Task t2 = Assert.ThrowsAsync<AuthenticationException>(() => serverStream.AuthenticateAsServerAsync(serverOptions, CancellationToken.None)).ContinueWith(t =>
+                            {
+                                server.Dispose();
+                            }, TaskScheduler.Default);
```

Comment:
What's the purpose of the ContinueWith here?  At the very least `t.GetAwaiter().GetResult();` should be added after the `server.Dispose();` call, so that any exceptions from the antecedent task are propagated.

Reply:
Tested this with wrong exception type, and the Continuewith task gets run even if the previous tasks fails, so the test was not doing the right thing. Changed this to try catch block. Upon receiving the AuthenticationException the server.dispose should be run to disconnect the tcpclient and server connection.